src/Collection.ts

```diff
@@ -77,27 +79,26 @@ class Collection extends Resource {
 
     // eager loading of 'fetchAllUri' (e.g. parent for embedded collections)
     if (this.config.avoidNPlusOneRequests) {
-      return LoadingCollection.create(itemsLoaded)
+      return LoadingCollection.create<ItemType>(itemsLoaded)
 
       // no eager loading: replace each reference (Link) with a Resource (ResourceInterface)
     } else {
-      return LoadingCollection.create(itemsLoaded, this._replaceEntityReferences(array))
+      return LoadingCollection.create<ItemType>(itemsLoaded, this._replaceEntityReferences(array))
     }
   }
 
   /**
    * Replace each item in array with a proper Resource (or LoadingResource)
    */
-  private _replaceEntityReferences (array: Array<Link>): Array<ResourceInterface> {
-    return array
-      .filter(entry => isEntityReference(entry))
-      .map(entry => this.apiActions.get(entry.href))
+  private _replaceEntityReferences (array: Array<ItemType>): Array<ItemType> {
+    const links = array.filter(entry => isEntityReference(entry)) as unknown as Link[]
```

Comment:
why do you cast it as Link[], when you could know that it is a link array by only allowing Links?
why Link[] instead of Array<Link>?

Reply:
I was able to reduce it to as Array<Link>, but the predicate in the filter is somehow not strong enough?

Reply:
Solved it, should not be an arrow function.